### PR TITLE
Declaring entity identifier for message types

### PR DESCRIPTION
**Pull request by matiit**
Based on: https://github.com/utilitywarehouse/protoc-gen-uwentity/pull/4

Contents: 
 - [x] Support nested messages identifiers

Sometimes our messages only wrap other messages but we still would like them to have `GetEntityIdentifier()` for them. 
Normally we would just create another id field on the wrapping message, but that is unnecessary - it would be good to use the entityIdentifier from one of the messages that are wrapped. 
For example, we have an `Account` proto:  

```protobuf
message Account {
  string id = 1 [(uw.entity.v1.identifier) = true];
  string account_number = 2;
  // account_label is the preferred way of referring to an account in the UI and communications like emails.
  // for example: John Bloggs or Mr & Mrs Andrews
  string account_label = 3;
  google.protobuf.Timestamp join_date = 4;
  bool marketing_opted_in = 5;
  account.occupancy_type.v1.OccupancyType occupancy_type = 6 [(validate.rules).enum.defined_only = true];
  bool is_staff = 7;
}
```
And we also need an Event for when Account is created. 
We can now use Accounts' entityIdentifier as entityIdentifier for Event too:

```protobuf
message AccountCreatedEvent {
  account.v3.Account account = 1 [(uw.entity.v1.identifier) = true];
}
```

 
 - [x] Error message when nested entity has no identifier
 
image

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -119,17 +114,59 @@ func main() {
...
+			// check if the nested message has an identifier
+			for _, f := range field.Message.Fields {
+				if proto.GetExtension(f.Desc.Options().(*descriptorpb.FieldOptions), entitypb.E_Identifier).(bool) {
+					nestedIdentifier = true
+				}
+			}
+
+			if !nestedIdentifier {
+				return identifier{}, fmt.Errorf("nested messages must have an identifier on %s: %s: %w", entity, field.Desc.Kind(), errUnsupportedField)
+			}
```
Comment (Nesze) [inline main.go:155]: Think it's worth adding a test for this.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+protoc-gen-uwentity
+testgen/
```
Comment (heedson) [inline .gitignore:2]: Try removing this

File: main.go (hunk 1 above)
Comment (matiit) [reply main.go:155]: I'll try to think about how to add one for it. For now I've added screenshot explaining how that error can happen.

File: .gitignore (hunk 2 above)
Comment (matiit) [reply .gitignore]: totally worked!

File: main.go (hunk 1 above)
Comment (Nesze) [reply main.go:155]: hah, didn't realize it's not straightforward to programatically test

**Review by denisnikulin — APPROVED** [review]

**Review by heedson — APPROVED** [review]

**Pull request merged**